bugsnag-android-core/src/test/java/com/bugsnag/android/BreadcrumbSerializationTest.kt

```diff
@@ -0,0 +1,27 @@
+package com.bugsnag.android
+
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import org.junit.runners.Parameterized.Parameter
+import org.junit.runners.Parameterized.Parameters
+import java.util.Date
+
+@RunWith(Parameterized::class)
+internal class BreadcrumbSerializationTest {
+
+    companion object {
+        @JvmStatic
+        @Parameters
+        fun testCases() = generateSerializationTestCases(
+            "breadcrumb",
+            Breadcrumb("hello world", BreadcrumbType.MANUAL, mutableMapOf(), Date(0))
```

Comment:
Should this have non-null metadata test as well?

Reply:
This test has been ported from #606 which aimed to only add one test case each to start off with. I can add more test cases if that's desirable.

Reply:
Added a test case to sanity-check that the map is serialized.